docs/src/appendix/cheatcodes/mock_call_when.md

```diff
@@ -0,0 +1,43 @@
+# `mock_call_when`
+
+Cheatcodes mocking contract entry point calls:
+
+## `MockCallData`
+
+```rust
+pub enum MockCallData {
+    Any,
+    Values: Span<felt252>,
+}
+```
+
+`MockCallData` is an enum used to specify for which call data the contract entry point will be mocked.
+- `Any` mock the contract entry point for any call data.
+- `Values` mock the contract entry point only for this call data.
+
+## `mock_call_when`
+> `fn mock_call_when<T, impl TSerde: serde::Serde<T>, impl TDestruct: Destruct<T>>(
+>   contract_address: ContractAddress, function_selector: felt252, call_data: MockCallData, ret_data: T, n_times: u32
+> )`
+
+Mocks contract call to a `function_selector` of a contract at the given address, with the given call data, for `n_times` first calls that are made 
```

Comment:
Here and in other places

```suggestion
Mocks contract call to a `function_selector` of a contract at the given address, with the given calldata, for `n_times` first calls that are made 
```

Reply:
Source code updated to use `calldata` instead of `call data`